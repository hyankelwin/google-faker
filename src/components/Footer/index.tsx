import React from 'react';
import { Container } from './styles';
import packageJson from '../../../package.json';

interface FooterProps {
  page: string;
  isError?: boolean;
}

const Footer: React.FC<FooterProps> = ({ page, isError }) => (
  <Container page={page} isError={isError}>
    <p>Â©Google 2021</p>
    <p>Version {packageJson.version}</p>
  </Container>
);

export default Footer;
