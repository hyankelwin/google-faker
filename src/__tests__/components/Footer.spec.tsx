import { mount } from 'enzyme';
import Footer from '../../components/Footer';

describe('Footer component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render Footer component with success', () => {
      const props = {
        page: 'home',
      };

      const wrapper = mount(<Footer {...props} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should be able to render Footer component when isError is false', () => {
      const props = {
        page: 'search',
        isError: false,
      };

      const wrapper = mount(<Footer {...props} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
