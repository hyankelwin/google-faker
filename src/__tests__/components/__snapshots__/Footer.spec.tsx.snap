// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should be able to passed in all tests on index.tsx should be able to render Footer component when isError is false 1`] = `
"<Footer page=\\"search\\" isError={false}>
  <styled.div page=\\"search\\" isError={false}>
    <div className=\\"sc-aXZVg VCsdY\\">
      <p>
        ©Google 2021
      </p>
      <p>
        Version 
        0.1.0
      </p>
    </div>
  </styled.div>
</Footer>"
`;

exports[`Footer component should be able to passed in all tests on index.tsx should be able to render Footer component with success 1`] = `
"<Footer page=\\"home\\">
  <styled.div page=\\"home\\" isError={[undefined]}>
    <div className=\\"sc-aXZVg isTNyU\\">
      <p>
        ©Google 2021
      </p>
      <p>
        Version 
        0.1.0
      </p>
    </div>
  </styled.div>
</Footer>"
`;
